### PR TITLE
storag/stream: fix stream_flash_bytes_written() return value

### LINKED ISSUE #26150: storage/stream: flash_img_bytes_written() might returns more than number of payload bytes written.
flash_img_bytes_written() returns number of payload bytes written to flash rounded up to the flash alignment (so physical write size) instead of number of payload bytes written. This is the bug as module aim to isolate the user form HW details.

here the fix: #26148

### PR DESCRIPTION
**Pull request by nvlsianpu**
stream_flash_bytes_written() returned number o bytes physically
written to the flash. This number might be bigger than the requested
number as is aligned to the device write-block-size.

stream_flash_bytes_written() should return number of bytes
written requested by the user.

fixes #26150

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]
This looks like a sane approach to me. I wonder if we have in-tree code, that depends on the current behaviour.

Perhaps we should document this in the release notes?

**Comment by nvlsianpu** [comment]
> I wonder if we have in-tree code, that depends on the current behaviour.

No, we don't have.

File: include/storage/stream_flash.h [hunk 1]
```diff
@@ -105,6 +105,9 @@ size_t stream_flash_bytes_written(struct stream_flash_ctx *ctx);
  * @param data data to write
  * @param len Number of bytes to write
  * @param flush when true this forces any buffered data to be written to flash
+ *        Flush write should be the last write operation in any writes row for
+ *        given context (although it is not mandatory if total data size is
+ *        multiplier of the buffer size).
```
Comment (carlescufi) [inline include/storage/stream_flash.h:110]:
```suggestion
 *        A flush write should be the last write operation in a sequence of write operations for
 *        given context (although this is not mandatory if the total data size is
 *        a multiple of the buffer size).
```

**Comment by nvlsianpu** [comment]
@henrikbrixandersen I register issue for that, so it will be listed in coming release note automatically.

**Comment by jdascenzio** [comment]
Ok, this patch corrects more elegantly the update problem I had with canopen (#26077)! thanks

**Review by carlescufi — APPROVED** [review]

**Pull request merged**